### PR TITLE
Don't mangle source pixel when alpha-blending to 8bpp

### PR DESCRIPTION
**Pull request by NiLuJe**
I'm not quite sure what the bit twiddling was attempting to do, because
I'm not familiar with those low bitdepths, but it's almost definitely
wrong for 8bpp targets, where we do *NOT* want/need *any* quantization.

Dates back to 4c9e894b23a5af82ba5b8b7d0678d37a9da11cad

Re https://github.com/koreader/koreader/issues/6345

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/crengine/352)

### CODE REVIEW

**Comment by NiLuJe** [comment]
(And by "source", I mean the document's pixel, not the image's. Which is the inverse of what the code calls stuff (image is data, doc buffer is dst).

**Comment by NiLuJe** [comment]
I also checked the PNG decode stuff, and it should be packing/expanding/stripping stuff properly so we get proper output at all times, which was my original hunch of where something might have been going wonky ;).

**Comment by NiLuJe** [comment]
Okay, still untested, but I should finally have gotten a grip on it ;p.

**Comment by poire-z** [comment]
I'm glad you're mastering that part of the code, because I never will (and don't want to :) !
So, can't comment much, but thanks :)

**Comment by NiLuJe** [comment]
A bit rusty, but it sorta came back to me after staring at it a bit ;p.

**Comment by NiLuJe** [comment]
Fun fact: I lied. I am somewhat familiar with 4bpp, because Kindles. And the 4bpp code is taking a shortcut by always using the same nibble, it should be switching between high/low nibble depending on whether the pixel is odd or even.

I don't think anyone should really care about < 8bpp in 2020 though, so, eh' :D.

**Comment by poire-z** [comment]
(Feel free to merge, but at daytime hours ! Because when doing it late in the night, https://github.com/koreader/crengine/commit/2bc7aa98c98e0c0d4f4c38a206e719a63ccb5b13 might happen :) - when I woke up, it felt like someone has shitted in my living room :) - been living quite a bit in crengine commit log for some years :)
tldr: please shorten the squashed concatenated messages, at least strip the empty lines, thanks :)

**Comment by NiLuJe** [comment]
(I managed to *almost* not fuckup the commit message :D).

**Comment by geek1011** [comment]
> (I managed to almost not fuckup the commit message :D

That duplicate commit message ... :)

Fun fact: GitHub squash messages are different than the built-in git ones, which also include the date, the commit hash, and a whole lot more whitespace.

Also, slightly off topic, but regarding the EPUB3 series metadata from the other PR, that other method you mentioned involving refining the title *was* the standard in 3.0, but got removed with barely any adoption due to the issues it caused with title parsing, and the overall complexity of it. The new refines method is only *slightly* saner IMO, but it really is the best option overall for complex metadata. I haven't seen it in the wild too much, but I fix my books to use it.

And, even more off topic, but I wish all publishers would switch to making proper pure EPUB 3.2 books (EPUB 2 is full of legacy stuff, and EPUB 3.0 is a mess), as they're really much simpler to parse and reason about than EPUB 2. I admit the spec is a bit convoluted, but I found the best way to get familiar is to read all of it from start to finish.

**Comment by NiLuJe** [comment]
@geek1011: Yeah, I'd *probably* have noticed if I'd done it manually ;D.

Then again, I might only have removed the leading star, who knows :D.
